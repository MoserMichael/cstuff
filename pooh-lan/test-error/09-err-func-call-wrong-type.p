

sub triangle(a , b , c)
  return sqr( ~num a * a + b * b + c * c )
end

println( ~msg 'square of triangle' .. triangle( ~a 1 ~b 2 ~c 3 ) )


msg='a string'
arr=[]
c = 2
println( ~msg 'wrong arguments' .. triangle( ~a msg ~b arr ~c c ))

arr = triangle( ~a 1 ~b 2 ~c 3 )
