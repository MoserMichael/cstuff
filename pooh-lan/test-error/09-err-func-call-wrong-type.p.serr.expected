
test-error/09-err-func-call-wrong-type.p(13): Error: parameter ~a is called as string but defined as any type
     13 |:....+....:....+....:....+....:....+....:....+....:....+....:....+....:....+....
        |println( ~msg 'wrong arguments' .. triangle( ~a msg ~b arr ~c c ))
        |.............................................^^^^^^

test-error/09-err-func-call-wrong-type.p(13): Error: parameter ~b is called as array but defined as any type
     13 |:....+....:....+....:....+....:....+....:....+....:....+....:....+....:....+....
        |println( ~msg 'wrong arguments' .. triangle( ~a msg ~b arr ~c c ))
        |....................................................^^^^^^
Type checking failed
